geode-core/src/distributedTest/java/org/apache/geode/internal/cache/persistence/PersistentRecoveryOrderDUnitTest.java

```diff
@@ -881,6 +881,32 @@ public void testSplitBrain() {
     });
   }
 
+  @Test
+  public void testRecoverableSplitBrain() {
```

Comment:
I am afraid this test does not cover the code change you have in `PersistenceAdvisorImpl`.  I don't see the log message of line 537 in `PersistenceAdvisorImpl` in the test run. Nor did I see `ConflictingPersistentDataException` in the test run. In the test, `vm2` is always online. Perhaps this is not a scenario of split brain that you expect. Maybe I am missing something.

Reply:
This test runs successfully even without your code change in `PersistenceAdvisorImpl`. So either the test has to be modified or the fix on the product has to be updated.

Reply:
I fixed to enhance the test cases.